File: Episode01-Move_Player/Monogame/Player.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework.Input;
using MonoGame.Extended;

namespace Shmup
{
    internal static class Player
    {
        /// <summary>
        /// Static class as there is only 1 player
        /// </summary>
        #region Private class variables
        private static float Speed;
        private static RectangleF Rectangle;
        private static Color Colour;
        #endregion
        #region Public Methods
        public static void CreatePlayer(int w, int h, float speed, Color colour)
        {
            /// This method is used to set the player rectangle width, height and colour
            /// It is similar to a constructor used on a non-static cless
            /// C# does have a proper static constructor, called when the first reference to the class is made
            /// but it is not used here, to replicate Lua and Python ways of doing things.
            Rectangle = new RectangleF(0, 0, w, h);
            Rectangle.Position = new Vector2((Shared.WIDTH - Rectangle.Width) / 2, Shared.HEIGHT - Rectangle.Height - 10); // starting position
            Speed = speed;
            Colour = colour;
        }
        public static void Update(float dt, KeyboardState keyboardState)
        {
            if (keyboardState.IsKeyDown(Keys.Left) || keyboardState.IsKeyDown(Keys.A))  // check for left arrow or 'A'
            {
                Rectangle.X -= Speed * dt;
                if (Rectangle.X < 0)    // Is the rectangle off-screen to the left?
                    Rectangle.X = 0;
            }
            else if (keyboardState.IsKeyDown(Keys.Right) || keyboardState.IsKeyDown(Keys.D))  // check for right arrow or 'D'
            {
                Rectangle.X += Speed * dt;
                if (Rectangle.X > Shared.WIDTH - Rectangle.Width) // Is the rectangle off-screen to the right?
                    Rectangle.X = Shared.WIDTH - Rectangle.Width;
            }
        }
        public static void Draw(SpriteBatch spriteBatch)
        {
            // FillRectangle is from MonoGame.Extended
            spriteBatch.FillRectangle(rectangle:Rectangle, color:Colour);
        }
        #endregion
    }
}

File: Episode03-Bullets_And_Collisions/Love2D/player.lua
local Player = {} 	-- simple table for Player. As there is only one, class not required
local Shared = require "shared"
local Rectangle = require "lib.Rectangle"

function Player.setProperties(width, height, speed)
	--[[ static class equivalent of a constructor ]]
	Player.speed = speed
	Player.rect = Rectangle(Shared.WIDTH / 2 - width / 2, Shared.HEIGHT - height - 10, width, height)
end

function Player.update(keyboard, dt)
	if keyboard.isDown("left") or keyboard.isDown("a") then
		Player.rect:updatePosition(Player.speed * dt * -1, 0) 	-- move Player left
	end
	if keyboard.isDown("right") or keyboard.isDown("d")then	
		Player.rect:updatePosition(Player.speed * dt, 0) 		-- move Player right
	end
	if Player.rect.x < 0 then							-- check if Player x position is off-screen left side
		Player.rect:setX(0)								-- whoops! change it to 0: left side of screen
	end
	if Player.rect.right > Shared.WIDTH then					-- check if Player x position is off-screen right side (take Player width into account)
		Player.rect:setX(Shared.WIDTH - Player.rect.width)		-- whoops! change it to right side of screen, less the Player width		
	end
end

function Player.draw()
	love.graphics.setColor(Shared.GREEN)			-- change colour to green
	love.graphics.rectangle("fill", Player.rect.x, Player.rect.y, Player.rect.w, Player.rect.h)
end

return Player    